### DISCUSSION TITLE
Translating entity and attribute names

### DISCUSSION THREAD

**Discussion by m0wlheld** [pypolestar/polestar_api#22] [General]
What is required to translate entity names and attribute names, like "Battery Level"?

**Comment by dgomes** [comment]
https://developers.home-assistant.io/blog/2022/12/01/entity_translations

**Reply by m0wlheld** [reply]
Yeah, I've seen this, but I don't get it. What I did:

* in `sensor.py` I added a translation key to the Battery Level sensor and changed the name's attribute content (for debugging purposes):

```python
    PolestarSensorDescription(
        key="battery_charge_level",
        translation_key="polestar_api_battery_charge_level",
        name="Battery level (Py)",
        query="getBatteryData",
        field_name="batteryChargeLevelPercentage",
        unit=PERCENTAGE,
        round_digits=0,
        state_class=SensorStateClass.MEASUREMENT,
        device_class=SensorDeviceClass.BATTERY,
        max_value=None,
    )
```

* in `strings.json` I added an entity entry with a different translation value:

```json
  "entity": {
    "sensor": {
      "polestar_api_battery_charge_level": {
        "name": "Battery Level (strings)"
      }
    }
  },
```

* and finally I added the same part to `de.json` in the translations folder:

```json
    "entity": {
      "sensor": {
        "polestar_api_battery_charge_level": {
          "name": "Batteriestand (de)"
        }
      }
    }
```

After restarting HA (container), the entity's name in the frontend is displayed "Battery Level (Py)", so strings.json and de.json are obviously ignored.

**Comment by m0wlheld** [comment]
Could be impossible (as of yet) as noted on https://developers.home-assistant.io/docs/internationalization/custom_integration/

**Comment by leeyuentuen** [comment]
i suppose this can be closed? this has been available in beta version?

**Reply by m0wlheld** [reply]
But i don't see it, see #70
